ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -964,6 +995,109 @@ const triggerSet: TriggerSet<Data> = {
+      outputStrings: {
+        text: {
+          en: 'Blue',
+        },
+      },
+    },
+    {
+      id: 'DSR Left Eye Red tether',
+      type: 'Tether',
+      netRegex: NetRegexes.tether({ id: '0034' }),
+      condition: (data, matches) => matches.source === data.me,
+      // See note above on Right Eye Blue Tether.
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Red',
+        },
+      },
+    },
+    {
+      id: 'DSR Eyes Dive Cast',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '68C3', source: ['Right Eye', 'Left Eye'], capture: false }),
+      // One cast for each dive.  68C3 is the initial cast/self-targeted ability.
+      // 68C4 is the damage on players.
+      suppressSeconds: 1,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Dives Soon',
+        },
+      },
+    },
+    {
+      id: 'DSR Eyes Dive Counter',
+      type: 'Ability',
+      // TODO: should this call out who it was on? some strats involve the
+      // first dive targets swapping with the third dive targets.
+      netRegex: NetRegexes.ability({ id: '68C4', source: 'Nidhogg', capture: false }),
+      // One ability for each player hit.
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => output[`dive${data.diveCounter}`]!(),
+      run: (data) => data.diveCounter++,
+      outputStrings: {
+        dive1: Outputs.num1,
+        dive2: Outputs.num2,
+        dive3: Outputs.num3,
+        dive4: Outputs.num4,
+      },
+    },
+    {
+      id: 'DSR Eyes Steep in Rage',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '68BD', source: ['Right Eye', 'Left Eye'], capture: false }),
+      // Each of the eyes (if alive) will start this aoe.  It has the same id from each eye.
+      suppressSeconds: 1,
+      response: Responses.bigAoe('alert'),
+    },
+    {
+      id: 'DSR Right Eye Reminder',
+      type: 'StartsUsing',
+      // If the Right Eye is dead and the Left Eye gets the aoe off, then the Right Eye
+      // will be revived and you shouldn't forget about it.
+      netRegex: NetRegexes.startsUsing({ id: '68BD', source: 'Left Eye' }),
+      delaySeconds: (_data, matches) => parseFloat(matches.castTime),
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Kill Right Eye',
+        },
+      },
+    },
+    {
+      id: 'DSR Spear of the Fury Limit Break',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '62E2', source: 'Ser Zephirin', capture: false }),
+      condition: (data) => data.role === 'tank',
+      // This is a 10 second cast, and (from video) my understanding is to
+      // hit tank LB when the cast bar gets to the "F" in "Fury", which is
+      // roughly 2.8 seconds before it ends.
+      delaySeconds: 10 - 2.8,
```

Comment:
I feel like we had a conversation about some sort of issue with using functions in `delaySeconds` properties, but I can't remember what it was. Is that what you're going for here?

Reply:
Yeah, cast times are 0.3 seconds shorter in the log lines than they are in practice.  You can see that if it says 2.7 cast time, it'll be exactly 3 seconds between cast and ability.

Reply:
I guess I never looked closely enough for that. Makes sense then.